corehq/apps/domain/auth.py

```diff
@@ -104,6 +104,10 @@ def wrapper(request, *args, **kwargs):
             response = HttpResponse(status=401)
             response['WWW-Authenticate'] = 'Basic realm="%s"' % realm
             return response
+
+        if hasattr(view, 'is_location_safe'):
+            wrapper.is_location_safe = view.is_location_safe
+            # TODO: ^^^ Nice try. Not good enough. See corehq.apps.domain.decorators
```

Comment:
I think decorating `wrapper` with `@wraps(view)` will preserve attributes set on the view function (plus things will be a lot easier to debug if you ever try to interact with that view in a shell session)

Reply:
`@wraps` will only copy ` __name__, __module__ and __doc__` (https://docs.python.org/2/library/functools.html#functools.update_wrapper)

Reply:
I think this has the same issue as the initial code, this is called after the middleware finishes

Reply:
I think it also copies `__dict__` by default as part of `WRAPPER_UPDATES`:
```python
In [1]: from functools import wraps

In [2]: def my_decorator(f):
   ...:     @wraps(f)
   ...:     def wrapper(*args, **kwargs):
   ...:         return f(*args, **kwargs)
   ...:     return wrapper
   ...: 

In [3]: def my_bad_decorator(f):
   ...:     def wrapper(*args, **kwargs):
   ...:         return f(*args, **kwargs)
   ...:     return wrapper
   ...: 

In [4]: def set_foo(f):
   ...:     f.foo = 'bar'
   ...:     return f
   ...: 

In [5]: @my_decorator
   ...: @set_foo
   ...: def my_fn():
   ...:     return 3
   ...: 

In [6]: @my_bad_decorator
   ...: @set_foo
   ...: def my_bad_fn():
   ...:     return 3
   ...: 

In [7]: my_fn.foo
Out[7]: 'bar'

In [8]: my_bad_fn.foo
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-8-936c6937c7d3> in <module>()
----> 1 my_bad_fn.foo

AttributeError: 'function' object has no attribute 'foo'
```
Not sure it solves the problem here anyways, as Cal said.